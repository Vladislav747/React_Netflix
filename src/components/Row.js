import axios from '../axios';
import React, {useState, useEffect} from 'react';

function Row ({title, fetchUrl}){
    const [movies, setMovies] = useState([]);

    //run once when the row loads, and don't run again
    useEffect(() => {
       async function  fetchData() {
           const request = await axios.get(fetchUrl);
           setMovies(request.data.results);
           return request;
       }
       fetchData();
    }, [fetchUrl]);

    console.log(movies, "movies");
    return(
    <div className="row">
        <h2>{title}</h2>
        <div class="row__posters">
            {movies}
        </div>
    </div>
    )
}


export default Row;