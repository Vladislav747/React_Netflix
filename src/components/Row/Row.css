.row {
  margin-left: 20px;
  color: #fff;
}

.row__posters {
  display: flex;
  overflow-y: hidden;
  overflow-x: scroll;
  padding: 2em;
}

.row__posters::-webkit-scrollbar {
  display: none;
}

.row__poster {
  object-fit: contain;
  width: 100%;
  max-height: 20vh;
  margin-right: 10px;
  transition: transform 450ms;
  opacity: 0.8;
}

.row__poster:hover {
  transform: scale(1.08);
  opacity: 1;
  cursor: pointer;
}

.row_posterLarge {
  max-height: 40vh;
}

.row_posterLarge:hover {
  transform: scale(1.09);
}
