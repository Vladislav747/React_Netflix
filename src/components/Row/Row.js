import axios from "../../axios";
import React, { useState, useEffect } from "react";
import Youtube from "react-youtube";
import movieTrailer from "movie-trailer";

import "./Row.css";

function Row({ title, fetchUrl, isLargeRow }) {
  const [movies, setMovies] = useState([]);
  const [trailerUrl, setTrailerUrl] = useState([false]);

  let base_url = "https://image.tmdb.org/t/p/original";

  isLargeRow = (isLargeRow === undefined) ? false : true; 
  //run once when the row loads, and don't run again
  useEffect(() => {
    async function fetchData() {
      const request = await axios.get(fetchUrl);
      //check for not null elements
      request.data.results = request.data.results.filter(element => element.backdrop_path !== null);
      setMovies(request.data.results);
      return request;
    }
    fetchData();
  }, [fetchUrl]);

  const opts = {
    height: "390",
    width: "100%",
    playerVars: {
      autoplay: 1,
    }
  };

  const handleClick = (movie) => {
    if(trailerUrl){
      setTrailerUrl('');
    }else{
      movieTrailer(movie.name || '')
        .then((url) => {
          console.log(url[0], "url");
          if(url[0]){
            const urlParams = new URLSearchParams(new URL(url).search);
            setTrailerUrl(urlParams.get("v"));
          }else{
            setTrailerUrl(false);
          }
          
        })
        .catch((error) => console.log(error))
    }
  }

  return (
    <div className="row">
      <h2>{title}</h2>
      <div className="row__posters">
        {movies.map((movie) => (
          
          <img
            key={movie.id}
            onClick={() => handleClick(movie)}
            className={`row__poster ${isLargeRow && "row_posterLarge"}`}
            src={`${base_url}${
              isLargeRow ? movie.poster_path : movie.backdrop_path
            }`}
            alt={movie.name}
            
          />
        ))}
      </div>
      {console.log(trailerUrl, "trailerUrl")}
      {trailerUrl[0] && <Youtube videoId={trailerUrl} opts={opts} />}
    </div>
  );
}

export default Row;
