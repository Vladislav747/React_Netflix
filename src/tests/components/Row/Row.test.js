import renderer from "react-test-renderer";
import React from "react";

import Row from "../../../components/Row/Row";

describe("Render Row", () => {
  it("render correctly Row component", () => {
    const rowComponent = renderer.create(<Row />).toJSON();
    expect(rowComponent).toMatchSnapshot();
  });

  it("check children prop type", () => {
    var rowComponentTest = mount(<Row />);
    // expect(rowComponentTest.props().children).toBeObject();
  });
});
