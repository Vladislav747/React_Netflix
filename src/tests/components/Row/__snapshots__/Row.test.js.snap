// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Row render correctly Row component 1`] = `
<div
  className="row"
>
  <h2 />
  <div
    className="row__posters"
  />
</div>
`;
